src/libs/elektra/keyset.c

```diff
@@ -846,15 +847,21 @@ if (result >= 0)
  *
  * @param ks the keyset to work with
  * @param key the key to check
+ * @param options lookup options to use
  * @return position where the key is (>=0) if the key was found
  * @return -insertpos -1 (< 0) if the key was not found
  *    so to get the insertpos simple do: -insertpos -1
  * @see ksLookup() for retrieving the found key
  */
-ssize_t ksSearch (const KeySet * ks, const Key * key)
+ssize_t ksSearch (const KeySet * ks, const Key * key, elektraLookupOptions options)
 {
-	// TODO #4039 implement optimization
-	return ksSearchInternal (ks, key);
+	if (options & KDB_O_OPMPHM) {
+        // build the opmphm for the keyset
+		elektraLookupBuildOpmphm(ks);
```

Comment:
I think we need to check whether the OPMPHM is already build before this.

Reply:
I skipped the opmphm building as `ksSearch` was taking a constant keyset and `elektraLookupBuildOpmphm` would change it.